.github/workflows/npm-test.yml

```diff
@@ -0,0 +1,140 @@
+name: NPM test
+
+on:
+  workflow_call:
+
+jobs:
+  lint:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/setup-node@v3
+        with:
+          node-version: '16.x'
+      - run: npm ci --ignore-scripts
+      - run: npm run lint
+
+  test-unit:
+    runs-on: ${{ matrix.operating-system }}
+    strategy:
+      matrix:
+        operating-system:
+          - ubuntu-latest
+          - windows-latest
+        node-version:
+          - '14.2'
+          - '14.x'
+          - '16.0'
+          - '16.x'
+          - '18.0'
+          - '18.x'
+    timeout-minutes: 15
+    steps:
+      - name: Use Node.js ${{ matrix.node-version }}
+        uses: actions/setup-node@v3
+        with:
+          node-version: ${{ matrix.node-version }}
+      - name: Ensure line endings are consistent
+        run: git config --global core.autocrlf input
```

Comment:
I wonder why we have this. I remember adding this but I don't remember why anymore. I would think this makes more sense in the linting test. Do you have any idea? If not we can just keep it I guess 😄 .

Reply:
I think this is needed for the windows unit tests.